### PR TITLE
Add instructions and countdown

### PR DESCRIPTION
**Pull request by hzxcaryn**
### Changes
Added instructions screen and countdown screen before game starts, according to mockup.
Changed existing countdown as a common component for all games to use which manages its own state.

### How to test
Can go to [`/balloon-game`](https://deploy-preview-56--pinda-fun.netlify.com/balloon-game) to start the a balloon game to see it in action

### CODE REVIEW

**Comment by hzxcaryn** [comment]
LOL i'm an idiot I didn't see #50

**Comment by juxd** [comment]
can just use this one, works better than mine. going to test the timers on the different browsers

**Comment by hzxcaryn** [comment]
okay, I'll refactor a bit and set the padding for the font bug HAHA sorry! :C

File: Web/src/components/games/GameInstructions.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  const [isCompleted, setIsCompleted] = useState(false);
+
+  useEffect(() => {
+    /* eslint-disable consistent-return */
+    if (isCompleted && onComplete) {
+      onComplete();
+      return;
+    }
+
+    if (!seconds) {
```
Comment (indocomsoft) [inline Web/src/components/games/GameInstructions.tsx]: Is this equivalent to `seconds === 0 || seconds == null`?

File: Web/src/components/games/GameInstructions.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      return;
+    }
+
+    const intervalId = setInterval(() => {
```
Comment (indocomsoft) [inline Web/src/components/games/GameInstructions.tsx]: Why not use `setTimeout`? Then don't need to have a destructor right lol

File: Web/src/components/games/GameInstructions.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  seconds,
+  onComplete,
+}) => {
+  const [isCompleted, setIsCompleted] = useState(false);
+
+  useEffect(() => {
+    /* eslint-disable consistent-return */
+    if (isCompleted && onComplete) {
+      onComplete();
+      return;
```
Comment (indocomsoft) [inline Web/src/components/games/GameInstructions.tsx]:
```suggestion
      return undefined;
```

Then you don't need to disable `consistent-return`

File: Web/src/components/games/GameInstructions.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    if (!seconds) {
+      return;
+    }
+
+    const intervalId = setInterval(() => {
+      setIsCompleted(true);
+    }, seconds * 1000);
+    return () => clearInterval(intervalId);
+    /* eslint-enable consistent-return */
```
Comment (juxd) [inline Web/src/components/games/GameInstructions.tsx]:
```suggestion

```

File: Web/src/components/games/GameInstructions.tsx (hunk 3 above)
Comment (juxd) [inline Web/src/components/games/GameInstructions.tsx]:
```suggestion
      return undefined;
```

File: Web/src/components/games/GameInstructions.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+  useEffect(() => {
+    /* eslint-disable consistent-return */
+    if (isCompleted && onComplete) {
+      onComplete();
+      return;
+    }
+
+    if (!seconds) {
+      return;
```
Comment (juxd) [inline Web/src/components/games/GameInstructions.tsx]:
```suggestion
      return undefined;
```

File: Web/src/components/games/GameInstructions.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+const GameInstructions: React.FC<GameInstructionsProps> = ({
+  children,
+  title,
+  seconds,
+  onComplete,
+}) => {
+  const [isCompleted, setIsCompleted] = useState(false);
+
+  useEffect(() => {
+    /* eslint-disable consistent-return */
```
Comment (juxd) [inline Web/src/components/games/GameInstructions.tsx]:
```suggestion

```

File: Web/src/components/games/GameInstructions.tsx (hunk 1 above)
Comment (hzxcaryn) [reply Web/src/components/games/GameInstructions.tsx]: Oh the latter makes more sense! Thanks for pointing this out LOL js

File: Web/src/components/games/BalloonShake/index.tsx [hunk 7]
```diff
@@ -8,13 +8,15 @@ import {
...
-import GameCountdown from './GameCountdown';
 import GameDisplay from './GameDisplay';
 import GameResults from './GameResults';
+import BalloonShakeInstructions from './BalloonShakeInstructions';
 import GamePrep from './GamePrep';
+import Countdown from '../Countdown';
 
 const GAME_TIME = 20; // Total shake time given
-const COUNTDOWN_TIME = 4; // Total time to countdown
+const INSTRUCTIONS_TIME = 5; // Total time to read instructions
```
Comment (indocomsoft) [inline Web/src/components/games/BalloonShake/index.tsx]: Later on we probably should put all these constants in one file

Comment (hzxcaryn) [reply Web/src/components/games/BalloonShake/index.tsx]: I also agree!

**Pull request merged**